docs/src/content/en/reference/memory/Memory.mdx

```diff
@@ -1,305 +1,154 @@
-# Memory Class Reference
+---
+title: "Reference: Memory Class | Memory | Mastra Docs"
+description: "Documentation for the `Memory` class in Mastra, which provides a robust system for managing conversation history and thread-based message storage."
+---
 
-The `Memory` class provides a robust system for managing conversation history and thread-based message storage in Mastra. It enables persistent storage of conversations, semantic search capabilities, and efficient message retrieval. You must configure a storage provider for conversation history, and if you enable semantic recall you will also need to provide a vector store and embedder.
-
-## Basic Usage
-
-```typescript copy showLineNumbers
-import { Memory } from "@mastra/memory";
-import { Agent } from "@mastra/core/agent";
-
-const agent = new Agent({
-  memory: new Memory(),
-  ...otherOptions,
-});
-```
-
-## Custom Configuration
-
-```typescript copy showLineNumbers
-import { Memory } from "@mastra/memory";
-import { LibSQLStore, LibSQLVector } from "@mastra/libsql";
-import { Agent } from "@mastra/core/agent";
```

Comment:
what are the reasons for removing these sections on each page? I think they're pretty helpful, if we want to show dry references first we should move this stuff to the bottom or make sure it's in general memory docs

Reply:
At the top of the page there's a usage example which normally just shows the required parameters / recommended parameters so users can quickly copy and and paste a minimal setup that works. 

The after the properties table there's an extended example which is where we can dive into some of the parameters a bit more. 

I don't think i've removed anything, it's just moved around a bit.

Reply:
Hey @TylerBarnes, here's a PR for the parts i remove from the reference pages in this PR: https://github.com/mastra-ai/mastra/pull/7557